codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/http/RequestBindingGenerator.kt

```diff
@@ -201,23 +203,43 @@ class RequestBindingGenerator(
                 val memberSymbol = symbolProvider.toSymbol(memberShape)
                 val memberName = symbolProvider.toMemberName(memberShape)
                 val outerTarget = model.expectShape(memberShape.target)
+
                 ifSet(outerTarget, memberSymbol, "&_input.$memberName") { field ->
                     // if `param` is a list, generate another level of iteration
-                    listForEach(outerTarget, field) { innerField, targetId ->
-                        val target = model.expectShape(targetId)
-                        rust(
-                            "query.push_kv(${param.locationName.dq()}, ${
-                            paramFmtFun(writer, target, memberShape, innerField)
-                            });",
-                        )
-                    }
+                    paramList(outerTarget, field, param, writer, memberShape)
+                }
+                if (memberShape.hasTrait<RequiredTrait>()) {
+                    rust(
+                        """
+                        else {
+                            query.push_kv(${param.locationName.dq()}, "");
+                        }
```

Comment:
IIRC, the goal is to have this behavior match labels, which throw a validation error:
https://github.com/awslabs/smithy-rs/blob/7f40bbcb2c2c1ed4f8c5096490ad161b60e884a0/codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/http/RequestBindingGenerator.kt#L250-L258